### PR TITLE
[docs] Add FLAML for efficient hyperparameter optimization

### PR DESCRIPTION
**Pull request by qingyun-wu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ Next you may want to read:
...
 - [**Distributed Learning**](https://github.com/microsoft/LightGBM/blob/master/docs/Parallel-Learning-Guide.rst) and [**GPU Learning**](https://github.com/microsoft/LightGBM/blob/master/docs/GPU-Tutorial.rst) can speed up computation.
 - [**Laurae++ interactive documentation**](https://sites.google.com/view/lauraepp/parameters) is a detailed guide for hyperparameters.
-- [**Optuna Hyperparameter Tuner**](https://medium.com/optuna/lightgbm-tuner-new-optuna-integration-for-hyperparameter-optimization-8b7095e99258) provides automated tuning for LightGBM hyperparameters ([code examples](https://github.com/optuna/optuna/blob/master/examples/)).
+- [**Optuna Hyperparameter Tuner**](https://medium.com/optuna/lightgbm-tuner-new-optuna-integration-for-hyperparameter-optimization-8b7095e99258) ([code examples](https://github.com/optuna/optuna/blob/master/examples/)) and [**FLAML**](https://www.microsoft.com/en-us/research/project/fast-and-lightweight-automl-for-large-scale-data/articles/flaml-a-fast-and-lightweight-automl-library/) ([code examples](https://github.com/microsoft/FLAML/blob/main/notebook/flaml_lightgbm.ipynb)) provide automated tuning for LightGBM hyperparameters.
```
Comment (jameslamb) [inline README.md:45]: could you please put FLAML on its own line here?

File: docs/Parameters-Tuning.rst [hunk 2]
```diff
@@ -7,7 +7,7 @@ This page contains parameters tuning guides for different scenarios.
 
 -  `Parameters <./Parameters.rst>`__
 -  `Python API <./Python-API.rst>`__
--  `Optuna`_ for automated hyperparameter tuning
+-  `Optuna`_ and `FLAML <https://github.com/microsoft/FLAML>`__ for automated hyperparameter tuning
```
Comment (jameslamb) [inline docs/Parameters-Tuning.rst:10]: could you please put FLAML on its own line here?

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks very much! Looks like an interesting project. Please just see two small suggestions I left.

File: README.md (hunk 1 above)
Comment (StrikerRUS) [inline README.md:45]:
```suggestion
- [**Optuna Hyperparameter Tuner**](https://medium.com/optuna/lightgbm-tuner-new-optuna-integration-for-hyperparameter-optimization-8b7095e99258) ([code examples](https://github.com/optuna/optuna/tree/master/examples/lightgbm)) and [**FLAML**](https://www.microsoft.com/en-us/research/project/fast-and-lightweight-automl-for-large-scale-data/articles/flaml-a-fast-and-lightweight-automl-library/) ([code example](https://github.com/microsoft/FLAML/blob/main/notebook/flaml_lightgbm.ipynb)) provide automated tuning for LightGBM hyperparameters.
```

File: README.md [hunk 3]
```diff
@@ -58,6 +58,7 @@ Some old update logs are available at [Key Events](https://github.com/microsoft/
 
 External (Unofficial) Repositories
 ----------------------------------
+FLAML (AutoML library for efficient hyperparameter optimization): https://github.com/microsoft/FLAML
```
Comment (StrikerRUS) [inline README.md:61]: I don't think that "efficient" word in this sentence adds any value, but just enlarges the description which should be compact. I don't think there are "inefficient" libraries for hyperparameter optimization 😄 .
```suggestion

FLAML (AutoML library for hyperparameter optimization): https://github.com/microsoft/FLAML
```

File: docs/Parameters-Tuning.rst (hunk 2 above)
Comment (StrikerRUS) [inline docs/Parameters-Tuning.rst:10]: Please make this reference as external (copy `Optuna` format).

**Review by StrikerRUS — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (qingyun-wu) [reply README.md]: done. thanks!

File: docs/Parameters-Tuning.rst (hunk 2 above)
Comment (qingyun-wu) [reply docs/Parameters-Tuning.rst]: done

File: README.md (hunk 1 above)
Comment (qingyun-wu) [reply README.md]: revised. thanks!

File: README.md (hunk 3 above)
Comment (qingyun-wu) [reply README.md]: sure. revised! thanks :)

File: docs/Parameters-Tuning.rst (hunk 2 above)
Comment (qingyun-wu) [reply docs/Parameters-Tuning.rst]: revised. thanks!

File: README.md [hunk 4]
```diff
@@ -58,6 +59,7 @@ Some old update logs are available at [Key Events](https://github.com/microsoft/
 
 External (Unofficial) Repositories
 ----------------------------------
+FLAML (AutoML library for hyperparameter optimization): https://github.com/microsoft/FLAML
```
Comment (StrikerRUS) [inline README.md:62]: Please add one blank line after a heading.
```suggestion

FLAML (AutoML library for hyperparameter optimization): https://github.com/microsoft/FLAML
```

File: docs/Parameters-Tuning.rst [hunk 5]
```diff
@@ -7,6 +7,7 @@ This page contains parameters tuning guides for different scenarios.
 
 -  `Parameters <./Parameters.rst>`__
 -  `Python API <./Python-API.rst>`__
+-  `FLAML`_  for automated hyperparameter tuning
```
Comment (StrikerRUS) [inline docs/Parameters-Tuning.rst:10]:
```suggestion
-  `FLAML`_ for automated hyperparameter tuning
```

**Review by StrikerRUS — APPROVED** [review]
LGTM, very nice indeed! Just two nits below.
Good luck with the package!

**Review by jameslamb — APPROVED** [review]

**Pull request merged**